### DISCUSSION TITLE
VM web camera

### DISCUSSION THREAD

**Discussion by Ltkirkland** [paukstelis/octoprint_deploy#119] [Q&A]
I apologies right off the bat because I must have screwed something up. I am swapping my octoprint server over to my truenas scale VM sine my sd card on my pi died today. I can get everything set up and running with USB passthrough but for some reason my camera services wont run (camera wont load stream) and it looks like my udev is set to false.

Any suggestions on how to fix this?
P.S. wonderful script.

octoprint_deploy diagnostic information. Please provide ALL output for support help
**************************************
/etc/octoprint_deploy
**************************************
haproxy: true
octoexec: /home/chris/OctoPrint/bin/octoprint
octopip: /home/chris/OctoPrint/bin/pip
streamer: ustreamer
**************************************
/etc/octoprint_instances
**************************************
instance:Vyper port:5000 udev:false
**************************************
/etc/octoprint_cameras
**************************************
camera:cam_Vyper port:8001 udev:false
cat: /etc/udev/rules.d/99-octoprint.rules: No such file or directory
**************************************
/etc/udev/rules.d/99-octoprint.rules
**************************************
ls: cannot access '/dev/octo*': No such file or directory
ls: cannot access '/dev/cam*': No such file or directory
**************************************
● Vyper.service - The snappy web interface for your 3D printer
     Loaded: loaded (/etc/systemd/system/Vyper.service; enabled; preset: enabled)
     Active: active (running) since Wed 2023-09-06 21:34:20 PDT; 6min ago
   Main PID: 2712 (octoprint)
      Tasks: 18 (limit: 2298)
     Memory: 141.8M
        CPU: 6.950s
     CGroup: /system.slice/Vyper.service
             └─2712 /home/chris/OctoPrint/bin/python3 /home/chris/OctoPrint/bin/octoprint serve --config=/home/chris/.Vyper/config.yaml --basedir=/home/chris/.Vyper --port=5000

Sep 06 21:36:10 Octoprint octoprint[2712]: 2023-09-06 21:36:10,907 - octoprint.access.users - INFO - Logged in user: cmuller12
Sep 06 21:36:10 Octoprint octoprint[2712]: 2023-09-06 21:36:10,913 - octoprint.server.util.flask - INFO - Passively logging in user cmuller12 from ::ffff:192.168.68.72
Sep 06 21:36:10 Octoprint octoprint[2712]: 2023-09-06 21:36:10,913 - octoprint.access.users - INFO - Logged in user: cmuller12
Sep 06 21:36:10 Octoprint octoprint[2712]: 2023-09-06 21:36:10,919 - octoprint.server.util.flask - INFO - Passively logging in user cmuller12 from ::ffff:192.168.68.72
Sep 06 21:36:10 Octoprint octoprint[2712]: 2023-09-06 21:36:10,919 - octoprint.access.users - INFO - Logged in user: cmuller12
Sep 06 21:36:11 Octoprint octoprint[2712]: 2023-09-06 21:36:11,159 - octoprint.server.util.sockjs - INFO - New connection from client: ::ffff:192.168.68.72
Sep 06 21:36:11 Octoprint octoprint[2712]: 2023-09-06 21:36:11,182 - octoprint.server.util.flask - INFO - Passively logging in user cmuller12 from ::ffff:192.168.68.72
Sep 06 21:36:11 Octoprint octoprint[2712]: 2023-09-06 21:36:11,183 - octoprint.access.users - INFO - Logged in user: cmuller12
Sep 06 21:36:11 Octoprint octoprint[2712]: 2023-09-06 21:36:11,641 - octoprint.server.util.sockjs - INFO - User cmuller12 logged in on the socket from client ::ffff:192.168.68.72
Sep 06 21:36:12 Octoprint octoprint[2712]: 2023-09-06 21:36:12,359 - octoprint.plugins.tracking - INFO - Sent tracking event webui_load, payload: {'browser_name': 'Chrome', 'browser_version': '116.0.0.0', 'os_name': 'Windows', 'os_version': '10'}
None
None
{'bitrate': '10000k',
 'cleanTmpAfterDays': 7,
 'defaultWebcam': 'classic',
 'ffmpeg': None,
 'ffmpegCommandline': '{ffmpeg} -framerate {fps} -i "{input}" -vcodec '
                      '{videocodec} -threads {threads} -b:v {bitrate} -f '
                      '{containerformat} -y {filters} "{output}"',
 'ffmpegThreads': 1,
 'ffmpegThumbnailCommandline': '{ffmpeg} -sseof -1 -i "{input}" -update 1 -q:v '
                               '0.7 "{output}"',
 'ffmpegVideoCodec': 'libx264',
 'snapshotWebcam': 'classic',
 'timelapse': {'fps': 25,
               'options': {'capturePostRoll': None,
                           'interval': None,
                           'retractionZHop': None},
               'postRoll': 0,
               'type': 'off'},
 'timelapseEnabled': True,
 'watermark': True,
 'webcamEnabled': True}
**************************************
○ cam_Vyper.service
     Loaded: masked (Reason: Unit cam_Vyper.service is masked.)
     Active: inactive (dead)
**************************************
● haproxy.service - HAProxy Load Balancer
     Loaded: loaded (/lib/systemd/system/haproxy.service; enabled; preset: enabled)
     Active: active (running) since Wed 2023-09-06 21:34:20 PDT; 6min ago
       Docs: man:haproxy(1)
             file:/usr/share/doc/haproxy/configuration.txt.gz
   Main PID: 2724 (haproxy)
      Tasks: 9 (limit: 2298)
     Memory: 14.7M
        CPU: 210ms
     CGroup: /system.slice/haproxy.service
             ├─2724 /usr/sbin/haproxy -Ws -f /etc/haproxy/haproxy.cfg -p /run/haproxy.pid -S /run/haproxy-master.sock
             └─2728 /usr/sbin/haproxy -Ws -f /etc/haproxy/haproxy.cfg -p /run/haproxy.pid -S /run/haproxy-master.sock

Sep 06 21:34:20 Octoprint systemd[1]: Starting haproxy.service - HAProxy Load Balancer...
Sep 06 21:34:20 Octoprint haproxy[2724]: [NOTICE]   (2724) : New worker (2728) forked
Sep 06 21:34:20 Octoprint haproxy[2724]: [NOTICE]   (2724) : Loading success.
Sep 06 21:34:20 Octoprint systemd[1]: Started haproxy.service - HAProxy Load Balancer.

**Comment by paukstelis** [comment]
It is not clear why the service has been created but masked and udev is set to false. I would try removing the camera instance and trying to add it again. If it is the same result, you can try" `sudo systemctl unmask cam_Vyper`
